src/Extend/Model.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Extend;
+
+use Flarum\Extension\Extension;
+use Illuminate\Contracts\Container\Container;
+
+class Model implements ExtenderInterface
+{
+    private $modelClass;
+    private $dateCallbacks = [];
+    private $defaultAttributeCallbacks = [];
+    private $relationships = [];
+
+    /**
+     * @param string $modelClass The ::class attribute of the model you are modifying.
+     *                           This model should extend from \Flarum\Database\AbstractModel.
+     */
+    public function __construct(string $modelClass)
+    {
+        $this->modelClass = $modelClass;
+    }
+
+    /**
+     * Modify which attributes of this model are treated as dates.
+     *
+     * @param callable $callable
+     *
+     * The callable can be a closure or invokable class, and should accept:
+     * - \Flarum\User\User $user: the user in question.
+     * - array $dateAttributes: an array of attributes which are processed as dates.
+     *
+     * The callable should return:
+     * - array $dateAttributes: an array of attributes which are processed as dates.
+     */
+    public function configureDates(callable $callable)
```

Comment:
Honestly, I don't see the use-case for full-blown manipulation currently (and also not in our existing - bundled - extensions). Can we stick to `date($attributeName)` for now?

Reply:
Agreed, I put it like this to make it maximally compatible with the old system, but I've been leaning towards this too.